index.js

```diff
@@ -225,21 +259,41 @@ function Physics (mcData, world) {
     // Finally, apply block collisions (web, soulsand...)
     playerBB.contract(0.001, 0.001, 0.001)
     const cursor = new Vec3(0, 0, 0)
+    let onSoulSand = false
+    let onHoneyBlock = false
     for (cursor.y = Math.floor(playerBB.minY); cursor.y <= Math.floor(playerBB.maxY); cursor.y++) {
       for (cursor.z = Math.floor(playerBB.minZ); cursor.z <= Math.floor(playerBB.maxZ); cursor.z++) {
         for (cursor.x = Math.floor(playerBB.minX); cursor.x <= Math.floor(playerBB.maxX); cursor.x++) {
           const block = world.getBlock(cursor)
           if (block) {
             if (block.type === soulsandId) {
-              vel.x *= physics.soulsandSpeed
-              vel.z *= physics.soulsandSpeed
+              onSoulSand = true
+            } else if (block.type === honeyblockId) {
+              onHoneyBlock = false
             } else if (block.type === webId) {
               entity.isInWeb = true
+            } else if (block.type === bubblecolumnId) {
+              const down = !block.metadata
+              const aboveBlock = world.getBlock(cursor.offset(0, 1, 0))
+              const bubbleDrag = (aboveBlock && aboveBlock.type === 0 /* air */) ? physics.bubbleColumnSurfaceDrag : physics.bubbleColumnDrag
+              if (down) {
+                vel.y = Math.max(bubbleDrag.maxDown, vel.y - bubbleDrag.down)
+              } else {
+                vel.y = Math.min(bubbleDrag.maxUp, vel.y + bubbleDrag.up)
+              }
             }
           }
         }
       }
     }
+    if (onSoulSand) { // dont apply soulsand slowdown several times if standing on the edge of the block
```

Comment:
Soulsand slowdown is actually applied multiple time if standing between different blocks (seen in the code and tested in 1.13). Idk about honey block and if they changed it in later versions.

https://www.mcpk.wiki/wiki/BlockSoulSand

Reply:
in minecraft 1.16.1 the velocity is only changed by the block directly below you:
![image](https://user-images.githubusercontent.com/16208640/90066117-fd1f3300-dcf5-11ea-94bb-ab2a5048782b.png)
(With getBlockPos being the floored coordinates of your legs, which in most cases is air, returning a multiplier of 1 at which point it calles getVelocityAffectPos, which returns the floored coordinates of your legs, besides the y level being the minY of your bounding box - 0.5)

Reply:
will go back version by version until it is changed, and will add a `supportFeature` for it

Reply:
On minecraft 1.15 i found the same exact code, but on mc 1.14 i found the code in the link you pasted, so it was changed in 1.15, since were splitting them up, ill completely move the handling of soulsand and honey on 1.15-1.16 out of the nested loop, as they should only be checking the block directly below them anyways

Reply:
I confirm that 1.16 and 1.15 behave differently than 1.14. I checked in game by walking on the edge of soulsand row: you don't get slowed down anymore. It confirms what you saw in code.